### PR TITLE
[Bugfix:TAGrading] Show graders/date not showing

### LINKED ISSUE #9891: Show graders/dates is buggy.
**Describe the bug**
Show graders/date option is buggy when grading labs.
image
If you are checking off a new question when show graders or/and show dates option is selected, the new question being checked off isn't showing any information as others.
Notice this bug also occurs reversely. First leave both show graders and show dates unselected,then check off a new question. Select either show graders or show dates at this point is still not going to give any information until refreshing.

**Expected behavior**
Informations should be revealed right after selecting show graders/dates.

**To Reproduce**
1. go to /courses/f23/sample/gradeable/grading_lab/grading?view=&sort=section_subsection
2. click show graders/dates
3. check off a new question
4. See error(Also do step 2 and 3 reversely)

**Configuration**
 - OS: [e.g. Windows, Linux, ...]

**Additional context**

### PR DESCRIPTION
**Pull request by ZheyuDeng**
Fix #9891 

### What is the current behavior?

When grading a lab, grader and date information will not appear on questions you have graded until you refresh the page. If you are checking off a new question when show graders or/and show dates option is selected, the new question being checked off does not show the same information as others.
This bug also occurs reversely.
image

### What is the new behavior?
Use websocket to pass the information and update the cells in real time to avoid refreshing the page.

https://github.com/Submitty/Submitty/assets/122962727/dcba7710-7dda-4bf0-ae6b-0aec19e009b9

### CODE REVIEW

**Review by zacharymnp — CHANGES_REQUESTED** [review]
Remaining console.log() statements are filling the console with debugging info.

File: site/app/controllers/grading/SimpleGraderController.php [hunk 1]
```diff
@@ -281,7 +281,10 @@ public function save($gradeable_id) {
 
         $this->core->getQueries()->saveTaGradedGradeable($ta_graded_gradeable);
 
-        return JsonResponse::getSuccessResponse($return_data);
+        return JsonResponse::getSuccessResponse([
+            'data' => $return_data,
+            'date' => $this->core->getDateTimeNow()->format('c')
+        ]);
```
Comment (williamjallen) [inline site/app/controllers/grading/SimpleGraderController.php:287]: Why is the `data` field being used here?  `getSuccessResponse()` already adds a top-level data field, so it seems odd to structure the response with two nested `data` fields.

File: site/public/js/simple-grading.js [hunk 2]
```diff
@@ -1,6 +1,35 @@
...
+    const showGraders = $('#show-graders').is(':checked');
+    const showDates = $('#show-dates').is(':checked');
+    $('.cell-grade').each(function() {
+        const graderElement = $(this).find('.simple-grade-grader');
+        const dateElement = $(this).find('.simple-grade-date');
+        if (showGraders && graderElement.text().trim() !== '') {
+            graderElement.css('display', 'block');
+        }
+        else {
+            graderElement.css('display', 'none');
```
Comment (williamjallen) [inline site/public/js/simple-grading.js:14]: This can be done with `.hide()`.  Also apply below.

File: site/public/js/simple-grading.js [hunk 3]
```diff
@@ -1,6 +1,35 @@
...
+        else {
+            dateElement.css('display', 'none');
+        }
+
+        // Force repaint by getting and setting the current color value
+        const graderColor = graderElement.css('color');
+        const dateColor = dateElement.css('color');
+
+        graderElement.css('color', graderColor);
+        dateElement.css('color', dateColor);
```
Comment (williamjallen) [inline site/public/js/simple-grading.js:29]: Is this necessary?

File: site/public/js/simple-grading.js [hunk 4]
```diff
@@ -564,7 +604,6 @@ function setupNumericTextCells() {
                             },
                         );
                     }
-
```
Comment (williamjallen) [inline site/public/js/simple-grading.js:567]: This whitespace change should be reverted.

**Review by williamjallen — CHANGES_REQUESTED** [review]
Minor changes, and then this should be ready to go

File: site/public/js/simple-grading.js (hunk 3 above)
Comment (ZheyuDeng) [reply site/public/js/simple-grading.js:29]: Yes it doesn't show without the repaint

**Review by williamjallen — APPROVED** [review]
Code seems reasonable to me.  Did not test manually.

File: site/public/js/simple-grading.js [hunk 5]
```diff
@@ -242,34 +275,41 @@ function updateCheckpointCells(elems, scores, no_cookie) {
...
+                        'date': formattedDate,
+                    });
+                }
+                else {
+                    console.log('Invalid date received:', returned_data.data.date);
+                }
             }
             else {
-                console.log('Save error: returned data:', returned_vals, 'does not match expected new data:', expected_vals);
+                console.log('Date not found in response:', returned_data);
```
Comment (ziesski) [inline site/public/js/simple-grading.js:305]: just wondering should we switch to more robust log system in near future? I looked up and seems like `loglevel` is commonly used as logging library for js. @williamjallen

**Review by zacharymnp — APPROVED** [review]
Functionality works.

File: site/public/js/simple-grading.js (hunk 5 above)
Comment (williamjallen) [reply site/public/js/simple-grading.js:305]: We don't really log much information except for in various error cases, so I don't really see a need to add a full JS logging library.

**Pull request merged**